src/coreclr/src/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -205,7 +205,11 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
     emitAttr emitSize;
     insOpts  opt = INS_OPTS_NONE;
 
-    if ((intrin.category == HW_Category_SIMDScalar) || (intrin.category == HW_Category_Scalar))
+    if (intrin.category == HW_Category_SIMDScalar)
```

Comment:
The changes here highlight the fact that there are many subtleties with regard to emit sizes - the "actual type" of the baseType, the "raw" (`emitTypeSize`) of the baseType, and the `emitSize` of the node itself. I think it would be worth some additional comments both here and in the cases where we use `emitTypeSize(node)` for moves.

Reply:
Agree, I will follow up and add the comments